core/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -298,8 +304,24 @@ public GetResult extractGetResult(final UpdateRequest request, String concreteIn
             }
         }
 
+        BytesReference sourceFilteredAsBytes = sourceAsBytes;
+        if (request.fetchSource() != null) {
```

Comment:
What about the case when `request.fetchSource().fetchSource()` is false? Why do we even have a boolean there? Maybe we should just use null instead? It doesn't make sense to have a `FetchSourceContext` with `fetchSource = false` and `includes = ["something"]`.